lib/imgix/path.rb

```diff
@@ -12,16 +12,14 @@ def initialize(prefix, secure_url_token, path = "/")
       @secure_url_token = secure_url_token
       @path = path
       @options = {}
-
-      @path = CGI.escape(@path) if /^https?/ =~ @path
-      @path = "/#{@path}" if @path[0] != "/"
     end
 
     def to_url(opts = {})
+      sanitized_path ||= sanitize_path(@path)
       prev_options = @options.dup
       @options.merge!(opts)
```

Comment:
Not related to this PR, but it would be great if this state mutation could be removed by passing state in method calls, like you're doing with `sanitized_path`.